go/vt/sqlparser/sql.y

```diff
@@ -5171,6 +5215,205 @@ trim_type:
     $$ = TrailingTrimType
   }
 
+frame_units:
+  ROWS
+  {
+    $$ = FrameRowsType
+  }
+| RANGE
+  {
+    $$ = FrameRangeType
+  }
+
+frame_point_type:
+  CURRENT ROW
+  {
+    $$ = CurrentRowType
+  }
+| UNBOUNDED PRECEDING
+  {
+    $$ = UnboundedPrecedingType
+  }
+| UNBOUNDED FOLLOWING
+  {
+    $$ = UnboundedFollowingType
+  }
+
+argument_less_window_expr_type:
+  CUME_DIST
+  {
+    $$ = CumeDistExprType
+  }
+| DENSE_RANK
+  {
+    $$ = DenseRankExprType
+  }
+| PERCENT_RANK
+  {
+    $$ = PercentRankExprType
+  }
+| RANK
+  {
+    $$ = RankExprType
+  }
+| ROW_NUMBER
+  {
+    $$ = RowNumberExprType
+  }
+
+frame_point:
+  frame_point_type
+  {
+    $$ = &FramePoint{Type:$1}
+  }
+| frame_expression PRECEDING
+  {
+    $$ = &FramePoint{Type:ExprPrecedingType, Expr:$1}
+  }
+| frame_expression FOLLOWING
+  {
+    $$ = &FramePoint{Type:ExprFollowingType, Expr:$1}
+  }
+
+frame_expression:
+  NUM_literal
+  {
+    $$ = $1
+  }
+| interval_value
+  {
+    $$ = $1
+  }
+
+frame_clause_opt:
+  {
+    $$ = nil
+  }
+| frame_clause
+  {
+    $$ = $1
+  }
+
+frame_clause:
+  frame_units frame_point
+  {
+    $$ = &FrameClause{ Unit: $1, Start: $2 }
+  }
+| frame_units BETWEEN frame_point AND frame_point
+  {
+    $$ = &FrameClause{ Unit: $1, Start: $3, End: $5, IsBetween: true }
+  }
+
+window_partition_clause_opt:
+  {
+    $$= nil
+  }
+| PARTITION BY expression_list
+  {
+    $$ = $3
+  }
+
+window_spec:
+sql_id window_partition_clause_opt order_by_opt frame_clause_opt
+  {
+    $$ = &WindowSpecification{ Name: $1.String(), PartitionClause: $2, OrderClause: $3, FrameClause: $4}
+  }
+|
+window_partition_clause_opt order_by_opt frame_clause_opt
+  {
+    $$ = &WindowSpecification{ PartitionClause: $1, OrderClause: $2, FrameClause: $3}
+  }
```

Comment:
Why not club them with `sql_id_opt`

Reply:
As far as I remember, there was a s/r conflict occurring when I tried to do so. I will recheck once and let you know.